src/ompl/util/src/RandomNumbers.cpp

```diff
@@ -217,3 +238,297 @@ void ompl::RNG::eulerRPY(double value[3])
     value[1] = acos(1.0 - 2.0 * uni_()) - boost::math::constants::pi<double>() / 2.0;
     value[2] = boost::math::constants::pi<double>() * (-2.0 * uni_() + 1.0);
 }
+
+
+void ompl::RNG::uniformNormalVector(unsigned int n, double value[])
+{
+    // Typedef the container type for the distribution and generator
+    typedef boost::numeric::ublas::shallow_array_adaptor<double> container_type_t;
+
+    // Construct the Boost distribution and generator. These would ideally be at the class level, but the spherical distribution requires dimension at construction.
+    // Tell them to return the type double in a shallow_array_adaptor container:
+    boost::uniform_on_sphere<double, container_type_t> uniformSphereDist(n);
+    boost::variate_generator<boost::mt19937&, boost::uniform_on_sphere<double, container_type_t> > normalizedVector(generator_, uniformSphereDist);
+
+    // Create a uBLAS-vector-view of the C-style array without copying data
+    container_type_t rVector(n, value);
+
+    // Generate a random value, the variate_generator is returning a shallow_array_adaptor, which will modify the value array:
+    rVector = normalizedVector();
+}
+
+// See: http://math.stackexchange.com/a/87238
+void ompl::RNG::uniformInBall(double r, unsigned int n, double value[])
+{
+    // Draw a random point on the unit sphere
+    this->uniformNormalVector(n, value);
+
+    // Draw a random radius scale
+    double radiusScale = r*std::pow(this->uniformReal(0.0, 1.0), 1.0/static_cast<double>(n));
```

Comment:
we usually do not use "this->" (here and elsewhere).
please add spaces around * and /